cranelift/codegen/src/machinst/lower.rs

```diff
@@ -295,6 +336,108 @@ pub struct Lower<'func, I: VCodeInst> {
     vm_context: Option<Reg>,
 }
 
+/// How is a value used in the IR?
+///
+/// This can be seen as a coarsening of an integer count. We only need
+/// distinct states for zero, one, or many.
+///
+/// This analysis deserves further explanation. The basic idea is that
+/// we want to allow instruction lowering to know whether a value that
+/// an instruction references is *only* referenced by that one use, or
+/// by others as well. This is necessary to know when we might want to
+/// move a side-effect: we cannot, for example, duplicate a load, so
+/// we cannot let instruction lowering match a load as part of a
+/// subpattern and potentially incorporate it.
+///
+/// Note that a lot of subtlety comes into play once we have
+/// *indirect* uses. The classical example of this in our development
+/// history was the x86 compare instruction, which is incorporated
+/// into flags useres (selectif, trueif, branches) and can
+/// subsequently incorporate loads, or at least we would like it
+/// to. However, danger awaits: the compare might be the only user of
+/// a load, so we might think we can just move the load (and nothing
+/// is duplicated -- success!), except that the compare itself is
+/// codegenned in multiple places, where it is incorporated as a
```

Comment:
```suggestion
/// codegen-ed in multiple places, where it is incorporated as a
```
I don't even know if this is more grammatically correct.

Reply:
I went with `codegen'd`. Someday dictionaries will catch up with compiler engineer jargon and tell us the right answer; until then that seems right to me :-)